src/exception.rs

```diff
@@ -277,8 +293,8 @@ impl KjError {
 
     /// Adds source location information to this error.
     #[must_use]
-    pub fn with_location(mut self, file: &str, line: u32) -> Self {
-        self.file = Some(file.to_string());
+    pub fn with_location(mut self, file: String, line: u32) -> Self {
```

Comment:
breaking change because of this, so will have to land carefully

Reply:
What's the particular reason for this change?

Reply:
to avoid double-allocation when you have a String already.